### PR TITLE
DB confirmed flight

### LINKED ISSUE #214: db confirmed flight
Create or update schema for confirmed flight

### PR DESCRIPTION
**Pull request by Plouc314**
* Update `FlightSchema` to have the `ConfirmedFlight` field, these fields are optional if the flight status is not `CONFIRMED`
* Update `FlightTable` to handle `ConfirmedFlight`
* Add a confirmed in `DatabaseSetup`
* Update `FlightTableUnitTest`, `FlightsViewModelTest`

Close #214

### CODE REVIEW

File: app/src/main/java/ch/epfl/skysync/database/schemas/FlightSchema.kt [hunk 1]
```diff
@@ -25,33 +27,61 @@ data class FlightSchema(
...
+                status = FlightStatus.PLANNED,
+                numPassengers = model.nPassengers,
+                timeSlot = model.timeSlot,
+                date = DateUtility.localDateToDate(model.date),
+            )
+        is ConfirmedFlight ->
+            FlightSchema(
+                id = model.id,
+                flightTypeId = model.flightType.id,
+                balloonId = model.balloon?.id,
```
Comment (Prolas) [inline app/src/main/java/ch/epfl/skysync/database/schemas/FlightSchema.kt:68]: Here you can replace it with a dot call, as the balloon and basket are not null in ConfirmedFlight

File: app/src/main/java/ch/epfl/skysync/database/schemas/FlightSchema.kt [hunk 2]
```diff
@@ -25,33 +27,61 @@ data class FlightSchema(
...
+                vehicleIds = model.vehicles.map { it.id },
+                status = FlightStatus.CONFIRMED,
+                numPassengers = model.nPassengers,
+                timeSlot = model.timeSlot,
+                date = DateUtility.localDateToDate(model.date),
+                remarks = model.remarks,
+                color = model.color,
+                meetupTimeTeam = DateUtility.localTimeToString(model.meetupTimeTeam),
+                departureTimeTeam = DateUtility.localTimeToString(model.departureTimeTeam),
+                meetupTimePassenger = DateUtility.localTimeToString(model.meetupTimePassenger),
```
Comment (Prolas) [inline app/src/main/java/ch/epfl/skysync/database/schemas/FlightSchema.kt:79]: Why are all these times `String` and not `Date ?

**Review by Prolas — APPROVED** [review]
LGTM

File: app/src/main/java/ch/epfl/skysync/database/schemas/FlightSchema.kt (hunk 2 above)
Comment (Plouc314) [reply app/src/main/java/ch/epfl/skysync/database/schemas/FlightSchema.kt:79]: For simplicity, I haven't had time to consult the documentation to see how time is handled.

**Pull request merged**